src/node/constant_source.rs

```diff
@@ -201,23 +201,33 @@ impl AudioProcessor for ConstantSourceRenderer {
 
         let offset = params.get(&self.offset);
         let output_channel = output.channel_data_mut(0);
-        let mut current_time = scope.current_time;
-
-        output_channel
-            .iter_mut()
-            .zip(offset.iter().cycle())
-            .for_each(|(o, &value)| {
-                if current_time < self.start_time || current_time >= self.stop_time {
-                    *o = 0.;
-                } else {
-                    // as we pick values directly from the offset param which is already
-                    // computed at sub-sample accuracy, we don't need to do more than
-                    // copying the values to their right place.
-                    *o = value;
-                }
-
-                current_time += dt;
-            });
+
+        // fast path
+        if offset.len() == 1
+            && self.start_time <= scope.current_time
+            && self.stop_time > next_block_time
```

Comment:
I guess it could be `self.stop_time >= next_block_time`, if stop time == next block time we can still fill the current buffer with constant value, no?

Reply:
Hmm, I think it would then not correspond to the check below:
```
if current_time < self.start_time || current_time >= self.stop_time {
    *o = 0.;
}
```


The spec says:
An [AudioScheduledSourceNode](https://webaudio.github.io/web-audio-api/#AudioScheduledSourceNode) is said to be playing when its associated [BaseAudioContext](https://webaudio.github.io/web-audio-api/#BaseAudioContext)'s [currentTime](https://webaudio.github.io/web-audio-api/#BaseAudioContext) is greater or equal to the time the [AudioScheduledSourceNode](https://webaudio.github.io/web-audio-api/#AudioScheduledSourceNode) is set to start, and less than the time it’s set to stop.

So only non-zero output for `start <= time < stop`?

Reply:
> So only non-zero output for start <= time < stop?

Yup agreed, but not sure this is contradictory actually

if `self.stop_time == next_block_time`, last sample time for the current block is still below `self.stop_time`
then in first sample of next block `current_time == self.stop_time` so it passes the check you mention

Or am I missing something ?